fenix/app/src/main/java/org/mozilla/fenix/compose/ext/Modifier.kt

```diff
@@ -100,3 +115,110 @@ fun Modifier.thenConditional(
     } else {
         this
     }
+
+/**
+ * The Composable this modifier is tied to may appear first and be fully constructed to then be pushed downwards
+ * when other elements appear. This can lead to over-counting impressions with multiple such events
+ * being possible without the user actually having time to see the UI or scrolling to it.
+ */
+private const val MINIMUM_TIME_TO_SETTLE_MS = 1000
+
+/**
+ * Add a callback for when this Composable is "shown" on the screen.
+ * This checks whether the composable has at least [threshold] ratio of it's total area drawn inside
+ * the screen bounds.
+ * Does not account for other Views / Windows covering it.
+ *
+ * @param threshold The ratio of the total area to be within the screen bounds to trigger [onVisible].
+ * @param settleTime The amount of time to wait before calling [onVisible].
+ * @param onVisible Invoked when the UI is visible to the user.
+ */
+fun Modifier.onShown(
```

Comment:
Doing some quick research, this Modifier extension cannot be covered by a unit test and will need a UI test if we choose to add coverage here.

Reply:
Gotcha, do you think we can add it in this PR?

Reply:
We certainly could if you feel this is a requirement for landing the change. I don't feel strongly either way

Reply:
It would increase our confidence in this logic and allow for easy refactoring later on when we move away from `composed {}`